pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/AbstractNode.java

```diff
@@ -373,8 +378,7 @@ public Node getNthParent(final int n) {
     private static <T> void findDescendantsOfType(final Node node, final Class<T> targetType, final List<T> results,
                                                   final boolean crossFindBoundaries) {
 
-        for (int i = 0; i < node.getNumChildren(); i++) {
-            final Node child = node.getChild(i);
+        for (Node child : node.children()) {
```

Comment:
On 7.0.x, node streams use these methods internally to iterate. This means, iteration here cannnot use node streams. #2220 anyway makes the class internal and documents this. I'll fix this when merging on 7.0.